azurerm/resource_arm_log_analytics_linked_service.go

```diff
@@ -0,0 +1,178 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2015-11-01-preview/operationalinsights"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLogAnalyticsLinkedService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogAnalyticsLinkedServiceCreateUpdate,
+		Read:   resourceArmLogAnalyticsLinkedServiceRead,
+		Update: resourceArmLogAnalyticsLinkedServiceCreateUpdate,
+		Delete: resourceArmLogAnalyticsLinkedServiceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
+
+			"workspace_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"linked_service_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "automation",
+			},
+
+			"linked_service_properties": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_id": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+
+			// Exported properties
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmLogAnalyticsLinkedServiceCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).linkedServicesClient
+	ctx := meta.(*ArmClient).StopContext
+	log.Printf("[INFO] preparing arguments for AzureRM Log Analytics linked services creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+
+	workspaceName := d.Get("workspace_name").(string)
+	lsName := d.Get("linked_service_name").(string)
+
+	props := d.Get("linked_service_properties").(map[string]interface{})
+	resourceID := props["resource_id"].(string)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := operationalinsights.LinkedService{
+		// Name: &name,
```

Comment:
Could we remove this?

Reply:
Yeah... no excuses for that one. Should have been removed.